noodles-bgzf/src/gzi.rs

```diff
@@ -0,0 +1,79 @@
+//! A module for the [GZI Index]. A GZI index contains pairs of compressed and uncompressed offsets
+//! in a BGZF file. Values in the index are stored as little-endian 64-bit unsigned integers.
+//!
+//! [GZI Index]: http://www.htslib.org/doc/bgzip.html#GZI_FORMAT
+
+pub use self::reader::Reader;
+
+mod reader;
+
+/// A GZI Index contains a number of entries, representing pairs of compressed and uncompressed
+/// offsets in a BGZF file.
+#[derive(Debug, PartialEq)]
+pub struct Index {
```

Comment:
I suggested creating a type alias for `Index` as `Vec<(u64, u64)>` in https://github.com/zaeleus/noodles/issues/103#issuecomment-1190472420 since a wrapper like this doesn't improve the model, e.g., `number_entries` is redundant given it's the same as `offsets.len()`.

You could create a newtype, but I'm not sure if it'd be useful.

Reply:
Yeah, you're right. I think the only benefit might be wrapping `offsets.len()` in a function which represents the definition of GZI, e.g:
```rust
  pub fn number_entries(&self) -> u64 {
      self.0.len() as u64
  }
```
Although, that might be complicating things more than necessary.